server/api/user.py

```diff
@@ -139,10 +140,76 @@ def acceptTerms(self, params):
     return resp
 
 
+@access.user
+@describeRoute(
+    Description('Pre-create a new user and issue them an invite.')
+    .param('login', 'The user\'s requested login.')
+    .param('email', 'The user\'s email address.')
+    .param('firstName', 'The user\'s first name.')
+    .param('lastName', 'The user\'s last name.')
+    .param('validityPeriod', 'The number of days that the invite will remain valid.',
+           required=False, dataType='float', default=30.0)
+)
+@boundHandler(_sharedContext)
+def inviteUser(self, params):
+    Token = self.model('token')
+    User = self.model('user', 'isic_archive')
+
+    params = self._decodeParams(params)
+    self.requireParams(['login', 'email', 'firstName', 'lastName'], params)
+    if 'validityPeriod' in params:
+        try:
+            validityPeriod = params['validityPeriod']
+        except ValueError:
+            raise ValidationException('Validity period must be a number.', 'validityPeriod')
+    else:
+        validityPeriod = 30.0
+
+    currentUser = self.getCurrentUser()
+    User.requireAdminStudy(currentUser)
+
+    newUser = User.createUser(
+        login=params['login'],
+        password=None,
+        email=params['email'],
+        firstName=params['firstName'],
+        lastName=params['lastName']
+        # TODO: suppress sending emails, in case email verification setting is enabled, as
+        # email verification will also be done when the temporary password is exchanged
+    )
+
+    # TODO: disable user
+
+    token = Token.createToken(
+        newUser, days=validityPeriod,
+        scope=[TokenScope.TEMPORARY_USER_AUTH, TokenScope.EMAIL_VERIFICATION])
```

Comment:
TODO: When the password is reset, also update the user document to reflect that the email has been verified.